### PR TITLE
Sdkconfig

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: toolchains/esp32/sdkconfig.defaults [hunk 1]
```diff
@@ -74,19 +65,13 @@ CONFIG_MBEDTLS_SSL_IN_CONTENT_LEN=7800
 CONFIG_MBEDTLS_SSL_OUT_CONTENT_LEN=3700
 # CONFIG_MBEDTLS_HARDWARE_MPI is not set
 CONFIG_MBEDTLS_TLS_CLIENT_ONLY=y
-# CONFIG_MBEDTLS_SSL_PROTO_TLS1 is not set
-# CONFIG_MBEDTLS_SSL_PROTO_TLS1_1 is not set
-CONFIG_MBEDTLS_SSL_PROTO_DTLS=n
```
Comment (floitsch) [inline toolchains/esp32/sdkconfig.defaults:79]: these don't seem to exist on the right side anymore.
something to worry about?

File: toolchains/esp32/sdkconfig.defaults [hunk 2]
```diff
@@ -1,70 +1,61 @@
...
-# CONFIG_ESP_TASK_WDT is not set
-CONFIG_ESP32_WIFI_CACHE_TX_BUFFER_NUM=16
+# CONFIG_ESP_TASK_WDT_INIT is not set
+# CONFIG_ESP_DEBUG_OCDAWARE is not set
+# CONFIG_ESP_BROWNOUT_DET is not set
 # CONFIG_ESP32_WIFI_NVS_ENABLED is not set
 CONFIG_ESP32_WIFI_TASK_PINNED_TO_CORE_1=y
 # CONFIG_ESP32_WIFI_IRAM_OPT is not set
-# CONFIG_FMB_CONTROLLER_SLAVE_ID_SUPPORT is not set
-CONFIG_FMB_TIMER_PORT_ENABLED=y
```
Comment (floitsch) [inline toolchains/esp32/sdkconfig.defaults:57]: These seem to have disappeared on the right.

File: toolchains/esp32/sdkconfig.defaults [hunk 3]
```diff
@@ -1,70 +1,61 @@
...
 # CONFIG_ESP_HTTP_CLIENT_ENABLE_HTTPS is not set
 CONFIG_ESP_SLEEP_GPIO_RESET_WORKAROUND=y
-CONFIG_ESP_IPC_TASK_STACK_SIZE=1024
-# CONFIG_ESP_NETIF_TCPIP_ADAPTER_COMPATIBLE_LAYER is not set
+CONFIG_XTAL_FREQ_AUTO=y
 # CONFIG_ESP_PHY_CALIBRATION_AND_DATA_STORAGE is not set
+CONFIG_ESP_DEFAULT_CPU_FREQ_MHZ_240=y
 # CONFIG_ESP_INT_WDT is not set
-# CONFIG_ESP_TASK_WDT is not set
-CONFIG_ESP32_WIFI_CACHE_TX_BUFFER_NUM=16
```
Comment (floitsch) [inline toolchains/esp32/sdkconfig.defaults:52]: Gone on the right?

File: toolchains/esp32/sdkconfig.defaults [hunk 4]
```diff
@@ -1,70 +1,61 @@
...
 # CONFIG_ADC_CAL_LUT_ENABLE is not set
+# CONFIG_SPI_SLAVE_ISR_IN_IRAM is not set
+CONFIG_UART_ISR_IN_IRAM=y
 # CONFIG_ESP_ERR_TO_NAME_LOOKUP is not set
 CONFIG_ETH_SPI_ETHERNET_DM9051=y
 CONFIG_ETH_SPI_ETHERNET_W5500=y
 CONFIG_ETH_USE_OPENETH=y
 # CONFIG_ESP_HTTP_CLIENT_ENABLE_HTTPS is not set
 CONFIG_ESP_SLEEP_GPIO_RESET_WORKAROUND=y
-CONFIG_ESP_IPC_TASK_STACK_SIZE=1024
```
Comment (floitsch) [inline toolchains/esp32/sdkconfig.defaults:47]: gone?

File: toolchains/esp32/sdkconfig.defaults [hunk 5]
```diff
@@ -1,70 +1,61 @@
...
-CONFIG_ESP32_DEFAULT_CPU_FREQ_240=y
-# CONFIG_ESP32_SPIRAM_SUPPORT is not set
-# CONFIG_SPIRAM_IGNORE_NOTFOUND is not set
-# CONFIG_SPIRAM_USE_CAPS_ALLOC is not set
-# CONFIG_SPIRAM_CACHE_LIBNUMPARSER_IN_IRAM is not set
-# CONFIG_SPIRAM_CACHE_LIBCHAR_IN_IRAM is not set
-# CONFIG_SPIRAM_CACHE_LIBSTR_IN_IRAM is not set
-# CONFIG_ESP32_DEBUG_OCDAWARE is not set
-# CONFIG_ESP32_BROWNOUT_DET is not set
-CONFIG_ESP32_XTAL_FREQ_AUTO=y
```
Comment (floitsch) [inline toolchains/esp32/sdkconfig.defaults:38]: Was this removed, or has it been removed because it's the default anyway?

**Review by floitsch — APPROVED** [review]
LGTM.

File: toolchains/esp32/sdkconfig.defaults (hunk 2 above)
Comment (erikcorry) [reply toolchains/esp32/sdkconfig.defaults:57]: This is something to do with modbus - it's not there any more in the new esp-idf.

File: toolchains/esp32/sdkconfig.defaults (hunk 1 above)
Comment (erikcorry) [reply toolchains/esp32/sdkconfig.defaults:79]: It looks like MbedTLS doesn't have support for older things that TLS 1.2 any more.

File: toolchains/esp32/sdkconfig.defaults (hunk 4 above)
Comment (erikcorry) [reply toolchains/esp32/sdkconfig.defaults:47]: This is default.

File: toolchains/esp32/sdkconfig.defaults (hunk 3 above)
Comment (erikcorry) [reply toolchains/esp32/sdkconfig.defaults:52]: I brought this back, and reduced some other buffer counts to 16. Nicely spotted.

**Pull request merged**